bumble/profiles/aics.py

```diff
@@ -0,0 +1,482 @@
+            properties=Characteristic.Properties.READ
+            | Characteristic.Properties.NOTIFY,
+            permissions=Characteristic.Permissions.READ_REQUIRES_ENCRYPTION,
+            gain_settings=gain_settings,
+            gain_settings_unit=gain_settings_unit,
+            mute=mute,
+            gain_mode=gain_mode,
+        )
+        self.gain_settings_properties = GainSettingsProperties(
+            uuid=GATT_GAIN_SETTINGS_ATTRIBUTE_CHARACTERISTIC,
+            properties=Characteristic.Properties.READ,
+            permissions=Characteristic.Permissions.READ_REQUIRES_ENCRYPTION,
+            gain_settings_unit=gain_settings_unit,
+        )
+        self.audio_input_type = Characteristic(
+            uuid=GATT_AUDIO_INPUT_TYPE_CHARACTERISTIC,
+            properties=Characteristic.Properties.READ,
+            permissions=Characteristic.Permissions.READ_REQUIRES_ENCRYPTION,
+            value=b'',
+        )
+        self.audio_input_status = Characteristic(
+            uuid=GATT_AUDIO_INPUT_STATUS_CHARACTERISTIC,
+            properties=Characteristic.Properties.READ,
+            permissions=Characteristic.Permissions.READ_REQUIRES_ENCRYPTION,
+            value=bytes([self.audio_input_status_value]),
+        )
+        self.audio_input_control_point = AudioInputControlPoint(
+            uuid=GATT_AUDIO_INPUT_CONTROL_POINT_CHARACTERISTIC,
+            properties=Characteristic.Properties.WRITE,
+            permissions=Characteristic.Permissions.WRITE_REQUIRES_ENCRYPTION,
+            audio_input_state=self.audio_input_state,
+            gain_settings_properties=self.gain_settings_properties,
+        )
+        self.audio_input_description = AudioInputDescription(
+            uuid=GATT_AUDIO_INPUT_DESCRIPTION_CHARACTERISTIC,
+            properties=Characteristic.Properties.READ
+            | Characteristic.Properties.NOTIFY
+            | Characteristic.Properties.WRITE_WITHOUT_RESPONSE,
+            permissions=Characteristic.Permissions.READ_REQUIRES_ENCRYPTION
+            | Characteristic.Permissions.WRITE_REQUIRES_ENCRYPTION,
+            audio_input_description_value=audio_input_description_value,
+        )
+
+        super().__init__(
+            [
+                self.audio_input_state,
+                self.gain_settings_properties,
+                self.audio_input_type,
+                self.audio_input_status,
+                self.audio_input_control_point,
+                self.audio_input_description,
+            ]
+        )
+
+
+# -----------------------------------------------------------------------------
+# Client
+# -----------------------------------------------------------------------------
+class AICSServiceProxy(ProfileServiceProxy):
+    SERVICE_CLASS = AICSService
+
+    def __init__(self, service_proxy: ServiceProxy) -> None:
+        self.service_proxy = service_proxy
+
+        self.audio_input_state = PackedCharacteristicAdapter(
```

Comment:
I recommend raising a `gatt.InvalidServiceError` when the characteristic isn't found (rather than have an `IndexError` raised when accessing `service_proxy.get_characteristics_by_uuid(xxx)[0]`.
See `profiles/bass.py` for an example.

Reply:
Indeed that's cleaner. Just done.